clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/ResizeAsgAtomicOperationUnitSpec.groovy

```diff
@@ -110,8 +111,65 @@ class ResizeAsgAtomicOperationUnitSpec extends Specification {
     capacity(0, 2, 3)    | 1          | 2          | 3              || true
   }
 
+  @Unroll
+  void "should support resize requests with only some fields to update (min: #minSize, max: #maxSize, desired: #desiredCapacity)"() {
+    setup:
+    def description = new ResizeAsgDescription(
+      asgs: [[
+               serverGroupName: "myasg-stack-v000",
+               region         : "us-west-1",
+               capacity       : new ServerGroup.Capacity(min: minSize, max: maxSize, desired: desiredCapacity)
+             ]]
+    )
+    def operation = new ResizeAsgAtomicOperation(description)
+    operation.amazonClientProvider = mockAmazonClientProvider
+
+    when:
+    operation.operate([])
+
+    then:
+    1 * mockAutoScaling.describeAutoScalingGroups(_) >> new DescribeAutoScalingGroupsResult().withAutoScalingGroups(
+      new AutoScalingGroup().withAutoScalingGroupName("myasg-stack-v0001")
+    )
+
+    1 * mockAutoScaling.updateAutoScalingGroup(_) >> {
+      request ->
+        assert request.size() == 1
+        assert request[0].minSize == minSize
+        assert request[0].maxSize == maxSize
+        assert request[0].desiredCapacity == desiredCapacity
+    }
+
+    where:
+    minSize | maxSize | desiredCapacity
+    42      | null    | null
+    null    | 42      | null
+    null    | null    | 42
+    0       | null    | null
+    0       | 0       | 0    // not the same as (null, null, null)
+  }
+
+  void "skip resizing if all capacity fields are null"() {
```

Comment:
I think you could avoid the need for this whole test if you just added a `null | null | null` to the previous test. 

You could add a new column:

```
minSize | maxSize | desiredSize || expectedAutoScalingInvocations 
```

and then

```
expectedAutoScalingInvocations * mockAutoScaling. describeAutoScalingGroups(_) >> {
expectedAutoScalingInvocations * mockAutoScaling.updateAutoScalingGroup(_) >> {
...
}
```

The null/null/null test is certainly valid, this suggestion is just aimed at getting coverage without needing 20 or so additional lines of testing code.

Reply:
in bde71d874